Modules/_io/fileio.c

```diff
@@ -1074,22 +1074,26 @@ fileio_repr(fileio *self)
     PyObject *nameobj, *res;
 
     if (self->fd < 0)
-        return PyUnicode_FromFormat("<_io.FileIO [closed]>");
+        return PyUnicode_FromFormat(
+            "<%s [closed]>",
```

Comment:
Would it be good to have a test for different cases? Something like below in test_fileio.py to ensure C and Python implementations behave the same.

```python
def test_custom_repr(self):
    class CustomFileIO(self.FileIO):
        pass

    custom_file_io = CustomFileIO(TESTFN, 'w')
    custom_file_io.close()
    self.assertIn("CustomFileIO", repr(custom_file_io))
```

Reply:
This PR definitely needs tests such as the one proposed here.

Reply:
@taleinat @tirkarthi thanks for comments. I added some tests as suggested.